src/main/java/org/openrewrite/java/migrate/lang/var/UseVarForGenericsConstructors.java

```diff
@@ -169,10 +181,22 @@ private J.VariableDeclarations transformToVar(J.VariableDeclarations vd, List<Ja
          * @return semantically equal Expression
          */
         private static Expression typeToExpression(JavaType type) {
-            if (type instanceof JavaType.Class) { // easy just parse to identifier
+            if (type instanceof JavaType.Primitive) {
+                JavaType.Primitive primitiveType = JavaType.Primitive.fromKeyword(((JavaType.Primitive) type).getKeyword());
+                return new J.Primitive(Tree.randomId(), Space.EMPTY, Markers.EMPTY, primitiveType);
+            }
+            if (type instanceof JavaType.Class) {
                 String className = ((JavaType.Class) type).getClassName();
                 return new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, className, type, null);
             }
+            if (type instanceof JavaType.Array){
+                List<JRightPadded<Space>> dimensions = IntStream.of(StringUtils.countOccurrences(type.toString(), "[]"))
+                        .mapToObj(i -> Space.EMPTY)
+                        .map(JRightPadded::build)
+                        .collect(Collectors.toList());
```

Comment:
Nice that you thought about multi dimensional arrays!

Reply:
Thanks! But I consider counting "[]" as a hack. I wasn't able to find something like JavaType.Array#getDimensions. There is no way in the API, yet?

Reply:
Not that I know; I see you've logged https://github.com/openrewrite/rewrite/issues/3425 ; seems like a good idea to me unless we both missed something.